jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostGeneralSuite.scala

```diff
@@ -335,4 +336,35 @@ class XGBoostGeneralSuite extends FunSuite with PerTest {
       assert(XGBoost.isClassificationTask(params) == isClassificationTask)
     }
   }
+
+  test("training with saving temporary boosters") {
+    import DataUtils._
+    val eval = new EvalError()
+    val trainingRDD = sc.parallelize(Classification.train).map(_.asML)
+    val testSetDMatrix = new DMatrix(Classification.test.iterator)
+
+    val tmpPath = Files.createTempDirectory("model1").toAbsolutePath.toString
+    val paramMap = List("eta" -> "1", "max_depth" -> 2, "silent" -> "1",
+      "objective" -> "binary:logistic", "hdfs_tmp_path" -> tmpPath,
+      "saving_frequency" -> 2).toMap
+    val finalModel = XGBoost.trainWithRDD(trainingRDD, paramMap, round = 5,
+      nWorkers = numWorkers)
+    def error(model: XGBoostModel): Float = eval.eval(
+      model.booster.predict(testSetDMatrix, outPutMargin = true), testSetDMatrix)
+    val model4 = XGBoost.loadModelFromHadoopFile(s"$tmpPath/4.model")
+    val model8 = XGBoost.loadModelFromHadoopFile(s"$tmpPath/8.model")
```

Comment:
why is it 4 and 8?

Reply:
Because the version increment twice for each round. So if the temp models are saved in round 2 and round 4, the model will be named as "4.model" and "8.model".
See here for more details: https://github.com/dmlc/xgboost/blob/master/jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java#L112